src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Services/Implementation/DelegationContextHandler.cs

```diff
@@ -73,5 +71,25 @@ public XacmlResourceAttributes GetResourceAttributes(XacmlContextRequest request
             XacmlContextAttributes resourceContextAttributes = request.GetResourceAttributes();
             return GetResourceAttributeValues(resourceContextAttributes);
         }
+
+        /// <summary>
+        /// Gets the list of mainunits for a subunit
+        /// </summary>
+        /// <param name="mainUnitQuery">Query model for the list of subunit partyIds to check and retrieve mainunits for</param>
+        /// <returns>List of mainunits</returns>
+        public async new Task<List<MainUnit>> GetMainUnits(MainUnitQuery mainUnitQuery)
+        {
+            return await GetMainUnits(mainUnitQuery);
```

Comment:
Shouldn't this be base.GetMainUnits? Seems you're calling the same method recursively now.

Reply:
The compiler seems to figure it out as it's not doing a recursive call, but I agree it should be specified for readability.

It's too bad the IContextHandler interface is owned by ABAC, and so can't easily be expanded for additional public operations. Which is the main reason behind the need for a DelegationContextHandler interface to begin with.